app/services/BackgroundTaskService.js

```diff
@@ -0,0 +1,123 @@
+
+function executeTask() {
+  check_intersect = () => {
+    // This function is called once every 12 hours.  It should do several things:
+
+    console.log('Intersect tick entering', isPlatformiOS());
+    // this.findNewAuthorities(); NOT IMPLEMENTED YET
+
+    // Get the user's health authorities
+    GetStoreData('HEALTH_AUTHORITIES')
+      .then(authority_list => {
+        if (!authority_list) {
+          // DEBUG: Force a test list
+          // authority_list = [
+          //  {
+          //    name: 'Platte County Health',
+          //    url:
+          //      'https://raw.githack.com/tripleblindmarket/safe-places/develop/examples/safe-paths.json',
+          //  },
+          //];
+          return;
+        }
+
+        let name_news = [];
+
+        if (authority_list) {
+          // Pull down data from all the registered health authorities
+          for (let authority of authority_list) {
+            fetch(authority.url)
+              .then(response => response.json())
+              .then(responseJson => {
+                // Example response =
+                // { "authority_name":  "Steve's Fake Testing Organization",
+                //   "publish_date_utc": "1584924583",
+                //   "info_website": "https://www.who.int/emergencies/diseases/novel-coronavirus-2019",
+                //   "concern_points":
+                //    [
+                //      { "time": 123, "latitude": 12.34, "longitude": 12.34},
+                //      { "time": 456, "latitude": 12.34, "longitude": 12.34}
+                //    ]
+                // }
+
+                // Update cache of info about the authority
+                // TODO: Add an "info_newsflash" UTC timestamp and popup a
+                //       notification if that changes, i.e. if there is a newsflash?
+                name_news.push({
+                  name: responseJson.authority_name,
+                  news_url: responseJson.info_website,
+                });
+
+                // TODO: Look at "publish_date_utc".  We should notify users if
+                //       their authority is no longer functioning.)
+
+                IntersectSet(responseJson.concern_points, dayBin => {
+                  if (dayBin !== null) {
+                    PushNotification.localNotification({
+                      title: languages.t('label.push_at_risk_title'),
+                      message: languages.t('label.push_at_risk_message'),
+                    });
+                  }
+                });
+              });
+
+            SetStoreData('AUTHORITY_NEWS', name_news)
+              .then(() => {
+                // TODO: Anything after this saves?  Background caching of
+                //       news to make it snappy?  Could be a problem in some
+                //       locales with high data costs.
+              })
+              .catch(error =>
+                console.log('Failed to save authority/news URL list'),
```

Comment:
Probably, make sense to pass is "error" prop to log for understanding, which kind of error we have

Reply:
Makes sense, will add.